lib/rubocop/cop/style/trailing_body_on_method_definition.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for trailing code after the method definition.
+      #
+      # @example
+      #   # bad
+      #   def some_method; do_stuff
+      #   end
+      #
+      #   # good
+      #   def some_method
+      #     do_stuff
+      #   end
+      #
+      class TrailingBodyOnMethodDefinition < Cop
+        include AutocorrectAlignment
+
+        MSG = 'Method body goes below definition.'.freeze
+
+        def on_def(node)
+          return unless node.body
+          return unless trailing_body?(node)
+
+          add_offense(node)
+        end
+        alias on_defs on_def
+
+        private
+
+        def trailing_body?(node)
```

Comment:
Refactoring this logic and adding more tests - WIP.

Reply:
Ping me when you require a review. 🙂

Reply:
@Drenmi - I believe this is ready now. I think I've found an accurate way to determine if it has a `trailing_body`.

Reply:
This method is incoherent if the `node.body` condition that is now in `#on_def` isn't moved inside. 🙂